Source/WebKit/Platform/IPC/unix/ConnectionUnix.cpp

```diff
@@ -226,7 +240,22 @@ bool Connection::processMessage()
     if (messageInfo.isBodyOutOfLine())
         messageBody = oolMessageBody->mutableSpan();
 
+#if OS(ANDROID)
+    const size_t hardwareBufferCount = messageInfo.hardwareBufferCount();
+    Vector<RefPtr<AHardwareBuffer>, 2> hardwareBuffers(hardwareBufferCount);
+    for (size_t i = 0; i < hardwareBufferCount; ++i) {
+        AHardwareBuffer* hardwareBufferPtr { nullptr };
+        int result [[maybe_unused]] = AHardwareBuffer_recvHandleFromUnixSocket(socketDescriptor(), &hardwareBufferPtr);
+        RELEASE_ASSERT(result == 0);
```

Comment:
This is guaranteed? No EXXXX codes that could be returned? (No idea about the ahb API... just guessing)

Reply:
As pointed out by @carlosgcampos in a private chat, we'll need to add error handling here, I checked the [implementation](https://android.googlesource.com/platform/frameworks/native/+/4f463a6b1de9198963dc6aff74154a504ba3f8f6/libs/nativewindow/AHardwareBuffer.cpp#392) and it only handles `EINTR` by itself. At least we'll want to retry on `-EWOULDBLOCK` given that our sockets are always configured as non-blocking.